lib/functions.php

```diff
@@ -333,16 +333,38 @@ function check_config($config_file) {
 	$config->setServers($servers);
 
 	# Check the memory limit parameter.
-	if ((ini_get('memory_limit') > -1) && ini_get('memory_limit') < $config->getValue('session','memorylimit'))
-		system_message(array(
-			'title'=>_('Memory Limit low.'),
-			'body'=>sprintf('Your php memory limit is low - currently %s, you should increase it to atleast %s. This is normally controlled in /etc/php.ini.',
-				ini_get('memory_limit'),$config->getValue('session','memorylimit')),
-			'type'=>'error'));
-
+	$limit = memory_str_to_int(ini_get('memory_limit'));
+	if ($limit != -1) {
+		$threshold = memory_str_to_int($config->getValue('session','memorylimit'));
+		if ($limit < $threshold) {
+			system_message(array(
+				'title' => _('Memory Limit low.'),
+				'body' => sprintf('Your php memory limit is low - currently %s (%s), you should increase it to atleast %s (%s). This is normally controlled in /etc/php.ini.',
+					ini_get('memory_limit'), $limit, $config->getValue('session','memorylimit'), $threshold),
+				'type'=>'error'
+			));
+		}
+	}
 	return $config;
 }
 
+/**
+ * Converts shorthand memory notation string to an integer that represents the
+ * given amount in bytes (ie. "128M" -> 134217728).
+ *
+ * @param string|int $value
+ * @return int
+ */
+function memory_str_to_int($value) {
+	$value = trim(strtolower($value));
```

Comment:
Passing an int to strtolower will make it unhappy, you should cast it before passing it to memory_str_to_int

Reply:
Yeah actually I just messed up with the docstring  `@param string|int $value`, it should be  `@param string $value`. 

The function clearly "Converts shorthand memory notation **string** to an integer", it expects a string returned by `ini_get()`, ie. `$limit = memory_str_to_int(ini_get(memory_limit));`

As stated [here](https://www.php.net/manual/en/function.ini-get.php#refsect1-function.ini-get-notes) :

> ini_get() will return the exact string stored in the php.ini file and NOT its int equivalent.

Reply:
Oh okay, thanks for the quick reply